### ISSUE TITLE
Don't assert and fail when there is no satisfiable solution

### ISSUE THREAD

**Issue by bunelr** [bamos/gurobi.torch#3]
First of all, thanks for writing those bindings!

I wanted to use them to solve some feasibility problems from Torch (checking whether some convex domains defined by a set of linear inequalities were empty or not).

Observed:
If I give an infeasible problem, i receive the following error:

```
luajit: /home/rudy/workspace/temp/gurobi.torch/init.c:103: GT_solve: Assertion `!error' failed.
Aborted (core dumped)
```
The error comes from [here](https://github.com/bamos/gurobi.torch/blob/master/init.c#L103). I'm not too familiar with Gurobi (first time using it ) but my assumption would be that because the problem was proven to be infeasible, there is no `X` attribute to get, hence the error.

I have a workaround for it on my branch [here](https://github.com/bunelr/gurobi.torch/tree/infeasible) which seems to do the trick, at least for my use case and was wondering if this should be brought in.

My (almost) minimial working example:
```lua
local gurobi = require 'gurobi'
local torch = require 'torch'


local is_feasible = function(ineq_set)
  local num_var = ineq_set:size(2) - 1

  -- We only care about the feasibility
  local null_cost = torch.zeros(num_var)

  local env = gurobi.loadenv("")
  local model = gurobi.newmodel(env, "", null_cost)
  gurobi.addconstrs(model, ineq_set:narrow(2,1,num_var), 'GE', -ineq_set:select(2,num_var+1))

  local status, x = gurobi.solve(model)

  -- Gurobi returns:
  --    2 if a solution was found / optimisation was successful.
  --    3 if no solution was found.
  return (status == 2)
end


local test_lp_solver = function()

  local feasible = torch.Tensor({ -- x<=0, y<=0, x<=-5
      {-1,0,0},
      {0,-1,0},
      {-1,0,-5}
  })

  local infeasible = torch.Tensor({ -- x>=y, x>=0, x<=-2
      {1,-1,0},
      {1,0,0},
      {-1,0,-2}
  })

  is_feasible(feasible)
  assert(is_feasible(feasible),
         "This set of inequality should be feasible but it is not reported so.")
  assert(not is_feasible(infeasible),
         "This set of inequality should be infeasible but it is not reported so.")
end

test_lp_solver()
```

Explanation for the status code was found from [Gurobi's optimization status page](http://www.gurobi.com/documentation/6.5/refman/optimization_status_codes.html)

Tangential note, something might be causing the issue: 
The version of Gurobi that I am using is 7.0 while these bindings seem to assume 6.5. (had to change a line [here](https://github.com/bamos/gurobi.torch/blob/master/init.lua#L13) to get it to point at the proper lib but otherwise everything seemed to work out of the box with the different version.)

Cheers,

**Comment by bamos** [comment]
Hi @bunelr - your code fixing this looks good, can you send in a PR?

-Brandon.

**Comment by bamos** [comment]
Resolved by #6
